lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -537,52 +537,88 @@ class AwsCompileFunctions {
+    } = functionObject;
+    let maximumEventAgeOnDestinations;
+    let maximumRetryAttemptsOnDestinations;
+
+    if (!destinations && !maximumEventAgeOnFunction && !maximumRetryAttemptsOnFunction) return;
+
+    const destinationConfig = {};
+
+    if (destinations) {
+      if (destinations.maximumRetryAttempts !== undefined) {
+        this.serverless._logDeprecation(
+          'AWS_FUNCTION_DESTINATIONS_MAXIMUM_RETRY_ATTEMPTS',
+          'destinations.maximumRetryAttempts is deprecated, use maximumRetryAttempts on function instead'
+        );
+        maximumRetryAttemptsOnDestinations = destinations.maximumRetryAttempts;
+      }
+
+      if (destinations.maximumEventAge !== undefined) {
+        this.serverless._logDeprecation(
+          'AWS_FUNCTION_DESTINATIONS_MAXIMUM_EVENT_AGE',
+          'destinations.maximumEventAge is deprecated, use maximumEventAge on function instead'
+        );
+        maximumEventAgeOnDestinations = destinations.maximumEventAge;
+      }
+
+      const hasAccessPoliciesHandledExternally = Boolean(
+        functionObject.role || this.serverless.service.provider.role
+      );
+      if (destinations.onSuccess) {
+        if (!hasAccessPoliciesHandledExternally) {
+          this.ensureTargetExecutionPermission(destinations.onSuccess);
+        }
+        destinationConfig.OnSuccess = {
+          Destination: destinations.onSuccess.startsWith('arn:')
+            ? destinations.onSuccess
+            : this.provider.resolveFunctionArn(destinations.onSuccess),
+        };
+      }
+      if (destinations.onFailure) {
+        if (!hasAccessPoliciesHandledExternally) {
+          this.ensureTargetExecutionPermission(destinations.onFailure);
+        }
+        destinationConfig.OnFailure = {
+          Destination: destinations.onFailure.startsWith('arn:')
+            ? destinations.onFailure
+            : this.provider.resolveFunctionArn(destinations.onFailure),
+        };
+      }
+    }
 
     const cfResources = this.serverless.service.provider.compiledCloudFormationTemplate.Resources;
     const functionLogicalId = this.provider.naming.getLambdaLogicalId(functionName);
+
+    const maximumEventAge = maximumEventAgeOnFunction || maximumEventAgeOnDestinations;
+    // For maximumRetryAttempts we cannot rely on "||" as the value can be Falsy, 0
+    const maximumRetryAttempts =
+      maximumRetryAttemptsOnFunction === undefined
+        ? maximumRetryAttemptsOnDestinations
+        : maximumRetryAttemptsOnFunction;
+
     const resource = {
       Type: 'AWS::Lambda::EventInvokeConfig',
       Properties: {
         FunctionName: { Ref: functionLogicalId },
-        DestinationConfig: {},
-        MaximumEventAgeInSeconds: destinations.maximumEventAge,
-        MaximumRetryAttempts: destinations.maximumRetryAttempts,
+        DestinationConfig: destinationConfig,
+        MaximumEventAgeInSeconds: maximumEventAge,
+        MaximumRetryAttempts: maximumRetryAttempts,
```

Comment:
We should set those properties only if provided.

While above is harmless for `undefined`, if someone put `null`, it'll break CF deployment, while in such case we should ensure that no such property is set